src/Chess/Game/ChessGame.php

```diff
@@ -1425,90 +1425,42 @@ public function in50MoveDraw()
-                    } else {
-                        return true;
-                    }
-                } else {
-                    return false;
-                }
-            } else {
-                if (in_array($whitePieces[0], array('N', 'B'))) {
-                    if (array_key_exists('N', $pieces['W']) && is_array($pieces['W']['N']) && count($pieces['W']['N']) > 1) {
-                        return false;
-                    } elseif (array_key_exists('B', $pieces['W']) && is_array($pieces['W']['B']) && count($pieces['W']['B']) > 1) {
-                        return false;
-                    } else {
-                        return true;
-                    }
-                } else {
-                    return false;
-                }
-            }
+        if ($piecesCount == 2) {
+            return true; //K vs K
         }
 
-        if (count($whitePieces) == 1) {
-            if (count($blackPieces) == 1) {
-                return true;
-            }
-            // XXX: The following if/else block appears to be unreachable due to the if/elseif/elseif block above
-            if ($blackPieces[0] == 'K') {
-                if (in_array($blackPieces[1], array('N', 'B'))) {
-                    if (is_array($pieces['B']['N']) && count($pieces['B']['N']) > 1) {
-                        return false;
-                    } elseif (is_array($pieces['B']['B']) && count($pieces['B']['B']) > 1) {
-                        return false;
-                    } else {
-                        return true;
-                    }
-                } else {
-                    return false;
-                }
-            } else {
-                if (in_array($blackPieces[0], array('N', 'B'))) {
-                    if (array_key_exists('N', $pieces['B']) && is_array($pieces['B']['N']) && count($pieces['B']['N']) > 1) {
-                        return false;
-                    } elseif (array_key_exists('B', $pieces['B']) && is_array($pieces['B']['B']) && count($pieces['B']['B']) > 1) {
-                        return false;
-                    } else {
-                        return true;
-                    }
-                } else {
-                    return false;
+        if ($piecesCount == 4) {
+            if (count($whitePieces) === count($blackPieces)) {
+                if ((in_array($whitePieces[0], array('N', 'B')) || in_array($whitePieces[1], array('N', 'B')))
+                && (in_array($blackPieces[0], array('N', 'B')) || in_array($blackPieces[1], array('N', 'B')))) {
+                    return true; //K+minor vs K+minor
                 }
             }
-        }
-        $wpIndex = ($whitePieces[0] == 'K') ? 1 : 0;
-        $bpIndex = ($blackPieces[0] == 'K') ? 1 : 0;
-        if ($whitePieces[$wpIndex] == 'B' && $blackPieces[$bpIndex] == 'B') {
-            // bishops of same color?
-            if ($pieces['B']['B'][0] == $pieces['W']['B'][0]) {
-                return true;
+
```

Comment:
Please add a test case to cover this scenario:

![2020-01-15-153808_3173x563_scrot](https://user-images.githubusercontent.com/45263/72461135-19cda980-37ad-11ea-8c89-54d4add53dba.png)

Reply:
Added